jadx-gui/src/main/java/jadx/gui/settings/JadxProject.java

```diff
@@ -179,9 +170,10 @@ public void saveAs(Path path) {
 	}
 
 	public void save() {
-		if (getProjectPath() != null) {
-			try (Writer writer = Files.newBufferedWriter(getProjectPath(), StandardCharsets.UTF_8)) {
-				GSON.toJson(data, writer);
+		Path savePath = getProjectPath();
+		if (savePath != null) {
+			try (Writer writer = Files.newBufferedWriter(savePath, StandardCharsets.UTF_8)) {
+				buildGson(savePath.getParent()).toJson(data, writer);
```

Comment:
```suggestion
				buildGson(savePath.toAbsolutePath().getParent()).toJson(data, writer);
```
@jpstotz you should be extra carefull while using `Path.getParent()` method because it can return `null` even for non-root files. For example `jadx-gui test.jadx` will fail with NPE. So it is better to expand path using `toAbsolutePath()` before calling `.getParent()`